controllers/external/util.go

```diff
@@ -39,9 +39,9 @@ func Get(ctx context.Context, c client.Reader, ref *corev1.ObjectReference, name
 	obj.SetAPIVersion(ref.APIVersion)
 	obj.SetKind(ref.Kind)
 	obj.SetName(ref.Name)
```

Comment:
I agree, passing in a namespace instead of using the one from the ref doesn't make sense. But I think we should also drop the additional namespace paramter. That is better then just changing the behavior and nobody will notice

Reply:
Should be updated now. Though it causes apidiff to fail, hope this is ok

Reply:
I think it's fine and actually that was my goal. 

I perfer having a breaking API change instead of only changing the behavior. Through the breaking change users have a chance to notice the behavioral change